lighthouse-core/audits/byte-efficiency/uses-responsive-images.js

```diff
@@ -141,21 +159,35 @@ class UsesResponsiveImages extends ByteEfficiencyAudit {
       // If naturalHeight or naturalWidth are falsy, information is not valid, skip.
       if (!naturalWidth || !naturalHeight) continue;
       const processed =
-        UsesResponsiveImages.computeWaste(
-          {...image, naturalHeight, naturalWidth},
-          ViewportDimensions, networkRecords
-        );
+         UsesResponsiveImages.computeWaste(
+           {...image, naturalHeight, naturalWidth},
+           ViewportDimensions, networkRecords
+         );
       if (!processed) continue;
 
+      // Set the allowed amount of waste per url
+      const existingAllowableWaste = allowableWasteMap.get(processed.url);
+      const computedAllowableWaste = this.computeAllowableWaste(image);
+      if (!existingAllowableWaste || existingAllowableWaste > computedAllowableWaste) {
+        allowableWasteMap.set(processed.url, computedAllowableWaste);
+      }
+
       // Don't warn about an image that was later used appropriately
       const existing = resultsMap.get(processed.url);
       if (!existing || existing.wastedBytes > processed.wastedBytes) {
         resultsMap.set(processed.url, processed);
       }
     }
 
+    // Ignore images that waste less than the allowed amount
     const items = Array.from(resultsMap.values())
-        .filter(item => item.wastedBytes > IGNORE_THRESHOLD_IN_BYTES);
+         .filter(item => {
+           let allowableWaste = allowableWasteMap.get(item.url);
+           if (!allowableWaste) {
```

Comment:
I think we can just move this filtering logic above, so there won't be a need to maintain a Map.

Reply:
I was assuming there was some reason that filter function was pulled out, but I guess there isn't a strong reason not to move it up. I've moved this filtering logic (as well as the logic I added) to the main loop.